web/util/queries.ts

```diff
@@ -96,18 +96,15 @@ const SPEAKER_WITH_SESSIONS = `
   _type,
   _updatedAt,
   "sessions": *[_type == "session" && references(^._id) && !(_id in path("drafts.**"))] {
-    _id,
-    title,
-    duration,
+    "session": { ${SESSION} },
```

Comment:
This should only query for the fields we actually need. If it helps, we could have a separate `SPEAKER_SESSION` const or something like that.

> the ${SESSION} query rewrite leads to more bytes queried, but this has no impact on the end user, as these pages are pre-generated

No, as far as I can tell, this is not the case. All this ends up being downloaded and processed by the client for "hydration" purposes. If I compare the size of the HTML page for Andy Fitzgerald's speaker page on `main` vs the `speaker-page-link-to-session` branch, it's well over 2 times as large now. The majority of it is a huge JSON blob at the end.

Reply:
2fe0413